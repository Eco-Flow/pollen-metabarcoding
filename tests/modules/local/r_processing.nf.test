nextflow_process {

    name "Test Process R_PROCESSING"
    script "modules/local/r_processing.nf"
    process "R_PROCESSING"

    test("Should run without failures") {

        when {
            params {
                clean = false
                outdir = "results"
            }
            process {
                """
                input[0] = Channel.of( [ [ "id":"ERR2537816", "single_end": false ], file("${projectDir}/data/processing-input/ERR2537816.tsv") ] ) 
                """
            }
        }

        then {
            assert process.success
            with (process.out.classification) {
              assert size() == 1
              with(get(0)) {
                    assert get(0) == [ "id": "ERR2537816", "single_end": false ]
                    assert get(1) ==~ ".*/ERR2537816.classified.tsv"
              }
            }
            with (process.out.pie) {
              assert size() == 1
              with(get(0)) {
                    assert get(0) == [ "id": "ERR2537816", "single_end": false ]
                    with(get(1)) {
                          assert get(0) ==~ ".*/ERR2537816.family.pdf"
                          assert get(1) ==~ ".*/ERR2537816.genus.pdf"
                          assert get(2) ==~ ".*/ERR2537816.order.pdf"
                    }
              }
            }
            with (process.out.summary_table) {
              assert size() == 1
              with(get(0)) {
                    assert get(0) == [ "id": "ERR2537816", "single_end": false ]
                    assert get(1) ==~ ".*/summary.tsv"
              }
            }
            assert snapshot(
              path("${launchDir}/${params.outdir}/r-processing/ERR2537816/ERR2537816.classified.tsv"),
              path("${launchDir}/${params.outdir}/r-processing/ERR2537816/summary.tsv")
            ).match("r_processing")
        }

    }

}
