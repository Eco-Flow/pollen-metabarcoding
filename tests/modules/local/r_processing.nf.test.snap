{
    "r_processing": {
        "content": [
            "ERR2537816.classified.tsv:md5,d7536874f239e304cca7667dc1ffde4d",
            "summary.tsv:md5,4a575d1bc4bfb5f79cf160ae793e9c30"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-27T13:36:37.712309"
    }
}